v2/pkg/cli/delete.go

```diff
@@ -0,0 +1,309 @@
+			ImageSetConfigurationSpec: v1alpha2.ImageSetConfigurationSpec{
+				Mirror: v1alpha2.Mirror{
+					Platform:         converted.Delete.Platform,
+					Operators:        converted.Delete.Operators,
+					AdditionalImages: converted.Delete.AdditionalImages,
+				},
+			},
+		}
+		o.Config = isc
+	}
+
+	// update all dependant modules
+	mc := mirror.NewMirrorCopy()
+	o.Manifest = manifest.New(o.Log)
+	o.Mirror = mirror.New(mc, nil)
+
+	// logic to check mode
+	// always good to check - but this should have been detected in validate
+	if strings.Contains(o.Opts.Global.DeleteSource, fileProtocol) {
+		wd := strings.Split(o.Opts.Global.DeleteSource, fileProtocol)
+		o.Opts.Global.WorkingDir = filepath.Join(wd[1], workingDir)
+	} else {
+		return fmt.Errorf("--source flag must have a file:// protocol prefix")
+	}
+
+	// setup logs level, and logsDir under workingDir
+	err := o.setupLogsLevelAndDir()
+	if err != nil {
+		return err
+	}
+	if o.isLocalStoragePortBound() {
+		return fmt.Errorf("%d is already bound and cannot be used", o.Opts.Global.Port)
+	}
+	o.LocalStorageFQDN = "localhost:" + strconv.Itoa(int(o.Opts.Global.Port))
+
+	// ensure mirror and batch worker use delete logic
+	o.Opts.Function = string(mirror.DeleteMode)
+	o.Log.Info("executing %s ", o.Opts.Function)
+
+	if o.Opts.Global.DeleteGenerate {
+		absPath, err := filepath.Abs(o.Opts.Global.WorkingDir + deleteDir)
+		if err != nil {
+			o.Log.Error("absolute path %v", err)
+		}
+		o.Log.Info("generate flag set, files will be created in %s", absPath)
+	}
+
+	if o.Opts.Global.ForceCacheDelete && !o.Opts.Global.DeleteGenerate {
+		o.Log.Info("force-cache-delete flag set, cache deletion will be forced")
+	}
+
+	if len(o.Opts.Global.DeleteID) > 0 {
+		o.Log.Info("using id %s to update all delete generated files", o.Opts.Global.DeleteID)
+	}
+
+	err = o.setupWorkingDir()
+	if err != nil {
+		return err
+	}
+	err = o.setupLocalStorageDir()
+	if err != nil {
+		return err
+	}
+
+	client, _ := release.NewOCPClient(uuid.New())
+	signature := release.NewSignatureClient(o.Log, o.Config, *o.Opts)
+	cn := release.NewCincinnati(o.Log, &o.Config, *o.Opts, client, false, signature)
+	o.Release = release.New(o.Log, o.LogsDir, o.Config, *o.Opts, o.Mirror, o.Manifest, cn, o.LocalStorageFQDN, o.ImageBuilder)
+	o.Batch = batch.New(o.Log, o.LogsDir, o.Mirror, o.Manifest)
+	// instantiate delete module
+	bg := archive.NewImageBlobGatherer(o.Opts)
+	o.Delete = delete.New(o.Log, *o.Opts, o.Batch, bg, o.Config, o.Manifest, o.LocalStorageDisk, o.LocalStorageFQDN)
+
+	return nil
+}
+
+// RunDelete - cobra run
+func (o *ExecutorSchema) RunDelete(cmd *cobra.Command) error {
+
+	startTime := time.Now()
+	o.Log.Debug("config %v", o.Config)
+	o.Log.Info(startMessage, o.Opts.Global.Port)
+
+	if o.Opts.Global.DeleteGenerate {
+		go startLocalRegistry(&o.LocalStorageService, o.localStorageInterruptChannel)
+
+		// lets get the release images from local disk
+		_, releaseFolder, err := o.Release.IdentifyReleases()
```

Comment:
To illustrate what I mean, I created this branch: https://github.com/sherine-k/oc-mirror/tree/CLID-20

Reply:
Yes this is in fact something I had in the back of my mind, I have created a FilterReleasesForDelete that uses semver and architecture to get the correct releases, agreed that IdentifyReleases() does not work as I thought it would